geopandas/tools/sjoin.py

```diff
@@ -233,6 +222,99 @@ def _geom_predicate_query(left_df, right_df, predicate):
     return indices
 
 
+def _reset_index_with_suffix(df, suffix, other):
+    """
+    Equivalent of df.reset_index(), but with adding 'suffix' to auto-generated
+    column names.
+    """
+    index_original = df.index.names
+    df_reset = df.reset_index()
+    column_names = df_reset.columns.values
+    for i, label in enumerate(index_original):
+        # if the original label was None, add suffix to auto-generated name
+        if label is None:
+            new_label = column_names[i]
+            if "level" in new_label:
+                # reset_index of MultiIndex gives "level_i" names, preserve the "i"
+                lev = new_label.split("_")[1]
+                new_label = f"index_{suffix}{lev}"
```

Comment:
This is a bit unfortunate, but previously we did some kind of our own auto-generation (instead of relying on the `reset_index` result), and so we used "index_right0", "index_right1" etc, while pandas uses "level_0", "level_1" etc. 
this part preserves this old naming scheme.